### PR TITLE
Migrate more tests to tests/docker

### PR DESCRIPTION
**Pull request by alex1545**
Migrating more tests to tests/docker

File existence tests:
* cc

Metadata tests:
* cc
* py
* py3
* go
* rust
* scala
* groovy

### CODE REVIEW

File: container/BUILD [hunk 1]
```diff
@@ -130,18 +130,11 @@ TEST_TARGETS = [
     ":cc_based_warning",
     "/pause_piecemeal:image",
     "/pause_piecemeal_gz:image",
-    ":cc_image",
```
Comment (nlopezgi) [inline container/BUILD]: Is it possible to fully remove the cc_image target and some of the below targets now from https://github.com/bazelbuild/rules_docker/blob/master/testdata/BUILD#L756?

File: tests/docker/groovy/configs/groovy_image.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+schemaVersion: 2.0.0
+
+metadataTest:
+  entrypoint: [
+    '/usr/bin/java',
+    '-cp',
+    '/app/io_bazel_rules_docker/tests/docker/groovy/libgroovy_image_library-impl.jar:/app/io_bazel_rules_docker/../com_google_guava_guava/jar/guava-18.0.jar:/app/io_bazel_rules_docker/../groovy_sdk_artifact/groovy-2.4.4/lib/groovy-2.4.4.jar:/app/io_bazel_rules_docker/tests/docker/groovy/libgroovy_image.binary-lib-impl.jar:/app/io_bazel_rules_docker/tests/docker/groovy/groovy_image.binary.jar:/app/io_bazel_rules_docker/tests/docker/groovy/groovy_image.binary:/app/io_bazel_rules_docker/tests/docker/groovy/BUILD',
```
Comment (xingao267) [inline tests/docker/groovy/configs/groovy_image.yaml]: Can this be broken into multilines?

Comment (alex1545) [reply tests/docker/groovy/configs/groovy_image.yaml]: I couldn't find a way to. All the ways to break a string into multilines in a YAML file will either insert a space or a new line char into the string.

Comment (xingao267) [reply tests/docker/groovy/configs/groovy_image.yaml]: Ack

**Comment by xingao267** [comment]
Check with @smukherj1 before merging. @smukherj1 is doing something with copybara.

**Comment by smukherj1** [comment]
Suggest waiting until my copybara CL is then. Then I suggest we use CaaS to sync a merged PR to internal codebara before merging any other PRs

**Comment by nlopezgi** [comment]
I think we can now fix the issues in this PR and get it submitted

**Pull request merged**